src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -133,7 +133,7 @@ object ResourceMatcher extends StrictLogging {
     * the reservation.
     */
   def matchResources(offer: Offer, runSpec: RunSpec, knownInstances: => Seq[Instance],
-    selector: ResourceSelector, conf: MatcherConf, schedulerPlugins: Seq[SchedulerPlugin] = Seq.empty)(implicit clock: Clock): ResourceMatchResponse = {
+    selector: ResourceSelector, conf: MatcherConf, schedulerPlugins: Seq[SchedulerPlugin] = Seq.empty, homeFaultDomain: Option[FaultDomain])(implicit clock: Clock): ResourceMatchResponse = {
```

Comment:
May be give it a default of `None` and avoid passing it altogether in tests later? Makes for a lot less changed lines.

Reply:
👍

Reply:
+1 this would make things a little simpler :).

Reply:
I agree that it would make it simpler, but in this case I'd make it explicit to not miss a single important part. If it looks too scary, can change it for sure :)